### PR TITLE
Refactor yargs (print help when no command; use commandDir)

### PR DESCRIPTION
**Pull request by dominykas**

### CODE REVIEW

**Review by rodion-arr — APPROVED** [review]
👍

File: USAGE.md [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
+
 
-Use this command to test your breaking changes against any one of your
-dependents. Wiby will go off to the dependent's repo and create a branch with a
-patch to the `package.json` pointing to your latest version (with the new
-changes) triggering the dependents CI to run.
+
+Use this command to fetch the results of your latest test against a dependent.
+Wiby will go off to the dependent's repo and fetch the results of the CI run
+against the patch branch Wiby had created.
```
Comment (ljharb) [inline USAGE.md:10]:
```suggestion
wiby will go off to the dependent’s repo and fetch the results of the CI run
against the patch branch wiby had created.
```
seems weird to capitalize it like it's a proper noun when it's an initialism :-p also, use a curly quote, straight quotes are always incorrect in prose.

File: USAGE.md [hunk 2]
```diff
@@ -12,15 +16,20 @@ Options:
...
 
-Use this command to fetch the results of your latest test against a dependent.
-Wiby will go off to the dependent's repo and fetch the results of the CI run
-against the patch branch Wiby had created
+## `wiby test`
+
+
+
+Use this command to test your breaking changes against any one of your
+dependents. Wiby will go off to the dependent's repo and create a branch with a
```
Comment (ljharb) [inline USAGE.md:25]:
```suggestion
dependents. wiby will go off to the dependent’s repo and create a branch with a
```

File: USAGE.md [hunk 3]
```diff
@@ -12,15 +16,20 @@ Options:
...
+patch to the  `package.json` pointing to your latest version (with the new
+changes) triggering the dependents CI to run.
```
Comment (ljharb) [inline USAGE.md:27]:
```suggestion
changes) triggering the dependent's CI to run.
```

File: bin/commands/result.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+'use strict'
+
+const wiby = require('../..')
+
+exports.desc = 'Use this command to fetch the results of your latest test against a dependent. Wiby will go off to the dependent\'s repo and fetch the results of the CI run against the patch branch Wiby had created.'
```
Comment (ljharb) [inline bin/commands/result.js:5]:
```suggestion
exports.desc = 'Use this command to fetch the results of your latest test against a dependent. wiby will go off to the dependent’s repo and fetch the results of the CI run against the patch branch wiby had created.'
```

File: bin/wiby [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      console.error(chalk.red.bold(msg));
+      console.error();
+    }
+
+    if (!err || err.showHelp) {
+      yargs.showHelp();
+    }
+
+    process.exit(1);
+  })
```
Comment (ljharb) [inline bin/wiby:29]: isn't this all automatic with `.help()` and `.strict()`?

File: USAGE.md (hunk 1 above)
Comment (dominykas) [reply USAGE.md]: https://github.com/pkgjs/wiby/pull/44/commits/62f833f769c41e29d7e86e794bead014b0d63abc

File: USAGE.md (hunk 2 above)
Comment (dominykas) [reply USAGE.md]: https://github.com/pkgjs/wiby/pull/44/commits/62f833f769c41e29d7e86e794bead014b0d63abc

File: bin/commands/result.js (hunk 4 above)
Comment (dominykas) [reply bin/commands/result.js]: https://github.com/pkgjs/wiby/pull/44/commits/62f833f769c41e29d7e86e794bead014b0d63abc

File: USAGE.md (hunk 3 above)
Comment (dominykas) [reply USAGE.md]: https://github.com/pkgjs/wiby/pull/44/commits/a50d9ac7c4ae8e4516bd58d2fe2a2f86e425f09a

File: bin/commands/test.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+'use strict'
+
+const wiby = require('../..')
+
+exports.desc = 'Use this command to test your breaking changes against any one of your dependents. Wiby will go off to the dependent\'s repo and create a branch with a patch to the  `package.json` pointing to your latest version (with the new changes) triggering the dependents CI to run.'
```
[the remark this answers is not in the document]
Comment (dominykas) [reply bin/commands/test.js]: https://github.com/pkgjs/wiby/pull/44/commits/62f833f769c41e29d7e86e794bead014b0d63abc, https://github.com/pkgjs/wiby/pull/44/commits/a50d9ac7c4ae8e4516bd58d2fe2a2f86e425f09a

File: bin/wiby (hunk 5 above)
Comment (dominykas) [reply bin/wiby:29]: `.help()` is the default and is therefore not strictly necessary.

`.strict()` does not play a part in this.

This custom `fail()` differs from the default handling in two aspects:

1. It does not print help (unless the error has a `showHelp` property) when the handler crashes. My experience is that the help, with a stack trace, just makes people confused.
2. It prints the error `msg` (don't ask me why Yargs uses two different args - `msg` and `err` for this) in red - this is usually the Yargs param validation message. It also prints it _above_ the help. Without this, in my experience, people just get confused and miss the error itself.

**Comment by dominykas** [comment]
@ljharb any further comments here or can we merge this?

**Review by ljharb — APPROVED** [review]

**Pull request merged**